ocpmodels/trainers/dist_forces_trainer.py

```diff
@@ -277,16 +285,19 @@ def train(self):
             self.model.train()
             for i, batch in enumerate(self.train_loader):
                 # Forward, loss, backward.
-                out = self._forward(batch)
-                loss = self._compute_loss(out, batch)
+                with torch.cuda.amp.autocast(enabled=self.scaler is not None):
+                    out = self._forward(batch)
+                    loss = self._compute_loss(out, batch)
+                loss = self.scaler.scale(loss) if self.scaler else loss
                 self._backward(loss)
+                scale = self.scaler.get_scale() if self.scaler else 1.
 
                 # Compute metrics.
                 self.metrics = self._compute_metrics(
                     out, batch, self.evaluator
                 )
                 self.metrics = self.evaluator.update(
-                    "loss", loss.item(), self.metrics
+                    "loss", loss.item() / scale, self.metrics
```

Comment:
We'd need an `all_reduce` call before this to compute the averaged loss correctly, right?

Reply:
Follow-up question to this:

An earlier version of DDP had https://github.com/Open-Catalyst-Project/baselines/blob/60ce4795c6d9c10ee0fa4f451ba8b18f51ec18fb/ocpmodels/trainers/dist_forces_trainer.py#L290-L301. Just to make sure we don't need that here as well as @abhshkdz noted.

Reply:
We only need this for reporting the train loss, but it's not that important (train loss should roughly be similar across nodes). We can add it if we want, but it will not change how training happens.